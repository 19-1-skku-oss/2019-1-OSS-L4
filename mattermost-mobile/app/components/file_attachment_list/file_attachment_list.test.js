// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import FileAttachment from './file_attachment_list.js';
import Preferences from 'mattermost-redux/constants/preferences';

jest.mock('react-native-doc-viewer', () => ({
    openDoc: jest.fn(),
}));

describe('PostAttachmentOpenGraph', () => {
    const loadFilesForPostIfNecessary = jest.fn().mockImplementationOnce(() => Promise.resolve({data: {}}));

    const baseProps = {
        actions: {
            loadFilesForPostIfNecessary,
        },
        canDownloadFiles: true,
        deviceHeight: 680,
        deviceWidth: 660,
        fileIds: ['fileId'],
        files: [{
            create_at: 1546893090093,
            delete_at: 0,
            extension: 'png',
            has_preview_image: true,
            height: 171,
            id: 'fileId',
            mime_type: 'image/png',
            name: 'image.png',
            post_id: 'postId',
            size: 14894,
            update_at: 1546893090093,
            user_id: 'userId',
            width: 425,
        }],
        postId: 'postId',
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot with a single image file', () => {
        const wrapper = shallow(
            <FileAttachment {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call loadFilesForPostIfNecessary when files does not exist', async () => {
        const props = {
            ...baseProps,
            files: undefined,
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>
        );
        expect(wrapper.state('loadingFiles')).toBe(true);
        expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledWith(props.postId);
        await loadFilesForPostIfNecessary();
        wrapper.setProps({files: baseProps.files});
        expect(wrapper.state('loadingFiles')).toBe(false);
    });

    test('should call loadFilesForPostIfNecessary on componentUpdate and when files does not exist', async () => {
        const loadFilesForPostIfNecessaryMock = jest.fn().mockImplementationOnce(() => Promise.resolve({data: {}}));
        const props = {
            ...baseProps,
            files: [],
            actions: {
                loadFilesForPostIfNecessary: loadFilesForPostIfNecessaryMock,
            },
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>
        );
        expect(wrapper.state('loadingFiles')).toBe(true);
        expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledTimes(1);
        expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledWith(props.postId);
        await loadFilesForPostIfNecessaryMock();
        expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledTimes(2);
        expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledWith(props.postId);
        await loadFilesForPostIfNecessaryMock();
        wrapper.setProps({files: baseProps.files});
        expect(wrapper.state('loadingFiles')).toBe(false);
    });
});
