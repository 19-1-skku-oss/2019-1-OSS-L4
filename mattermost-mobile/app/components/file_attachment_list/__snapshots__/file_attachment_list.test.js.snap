// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostAttachmentOpenGraph should match snapshot with a single image file 1`] = `
<ScrollViewMock
  horizontal={true}
  keyboardShouldPersistTaps="always"
  scrollEnabled={false}
  style={
    Array [
      undefined,
    ]
  }
>
  <FileAttachment
    canDownloadFiles={true}
    deviceWidth={660}
    file={
      Object {
        "caption": "image.png",
        "data": Object {
          "create_at": 1546893090093,
          "delete_at": 0,
          "extension": "png",
          "has_preview_image": true,
          "height": 171,
          "id": "fileId",
          "mime_type": "image/png",
          "name": "image.png",
          "post_id": "postId",
          "size": 14894,
          "update_at": 1546893090093,
          "user_id": "userId",
          "width": 425,
        },
      }
    }
    id="fileId"
    index={0}
    onCaptureRef={[Function]}
    onPreviewPress={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</ScrollViewMock>
`;
